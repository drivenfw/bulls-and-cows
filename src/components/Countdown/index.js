import React from 'react'
import PropTypes from 'prop-types'
import styled, { keyframes } from 'styled-components'


const pulse = props => keyframes`
  0% { 
    transform: scale(1);
    box-shadow: 0 0 0 0 ${props.theme.primaryColor1};
  }
  70% {
    transform: scale(1.1);
    box-shadow: 0 0 0 20px #fff;
  }
  100% {
    transform: scale(1);
    box-shadow: 0 0 0 0 #fff;
  }
`

const StyledCountdown = styled.div`
  width: 100px;
  height: 100px;
  border: 5px solid ${props => props.theme.primaryColor};
  border-radius: 50%;
  font-size: 2.5em;
  display: flex;
  align-items: center;
  justify-content: center;
  animation: ${pulse} 1500ms;
`

const Countdown = ({ value }) =>
  <StyledCountdown key={value}>
    {value}
  </StyledCountdown>

Countdown.propTypes = {
  value: PropTypes.number.isRequired
}

export default Countdown

