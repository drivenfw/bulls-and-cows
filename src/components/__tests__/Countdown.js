import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Countdown from 'components/Countdown'


configure({ adapter: new Adapter() })

describe('<Countdown />', () => {
  describe('snapshot test', () => {
    it('should match the snapshot', () => {
      const tree = shallow(<Countdown value={3} />)

      expect(toJson(tree)).toMatchSnapshot()
    })
  })
})

