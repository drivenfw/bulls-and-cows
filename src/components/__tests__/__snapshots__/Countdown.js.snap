// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Countdown /> snapshot test should match the snapshot 1`] = `
<styled.div
  key="3"
>
  3
</styled.div>
`;
